testsuite/cluster-test/src/prometheus.rs

```diff
@@ -100,9 +102,51 @@ impl Prometheus {
             .avg()
             .ok_or_else(|| format_err!("Failed to compute avg"))
     }
+
+    pub fn query_range_max(
+        &self,
+        query: String,
+        start: &Duration,
+        end: &Duration,
+        step: u64,
+    ) -> Result<HashMap<String, f64>> {
+        let response = self.query_range(query, start, end, step)?;
+        response
+            .max()
+            .ok_or_else(|| format_err!("Failed to compute max"))
+    }
 }
 
 impl MatrixResponse {
+    pub fn max(&self) -> Option<HashMap<String, f64>> {
+        if self.inner.is_empty() {
+            return None;
+        }
+        let mut max = HashMap::new();
+        let mut count = 0usize;
+        for (peer_id, time_series) in &self.inner {
+            let mut cur_max = 0.;
+            for (_, v) in time_series.inner.iter() {
+                if !v.is_normal() {
+                    // Some time series can return NaN
+                    // we simply ignore NaN values when calculating max
+                    continue;
+                }
+                if *v > cur_max {
+                    cur_max = *v;
+                }
+            }
+
+            max.insert(peer_id.clone(), cur_max);
+            count += 1;
+        }
+        if count == 0 {
```

Comment:
You may want a simple boolean for this guard. Or you could use the size of the max table.